Modules/Scripted/DICOM/DICOM.py

```diff
@@ -329,7 +343,24 @@ def setup(self):
     self.dicomFrame.setText("DICOM Database and Networking")
     self.layout.addWidget(self.dicomFrame)
 
-    self.detailsPopup = self.getSavedDICOMDetailsWidgetType()()
+    # Create widget and add a new DICOM browser layout
+    self.viewFactory = slicer.qSlicerSingletonViewFactory()
+    self.viewFactory.setTagName("dicombrowser")
+
+    self.detailsPopup = None
+    self.setDetailsPopup(self.getSavedDICOMDetailsWidgetType()())
+
+    layoutManager = slicer.app.layoutManager()
+    layoutManager.registerViewFactory(self.viewFactory)
+    self.DICOM_BROWSER_LAYOUT_ID = 2200
```

Comment:
It would be better to define this layout ID as a constant outside of the class so that you don't need to instantiate a class to access it.

Reply:
### the layout base id approach

For layout id, I am wondering if we should allocate 100 slots per extension or module or domain, and come up with hash function returning an integer and multiply it by 100:

```
unsigned int ComputeLayoutBaseId(const std::string& domain)
```

That way for any given module or extension, we could be sure to have a unique base ID and we would avoid clash.

layout specific to a module/extension would then be indexed relative to their layout base id.

### explicit registry

An other approach would be to allocate layout base id and keep track of assignment in a shared document.

Reply:
Choosing to use integer layout identifiers was a mistake. We cannot correct it in Slicer-4.10.x but we should switch to string IDs in Slicer5. String IDs can may be prefixed with modulename+"." as we do it for singleton tags and node attributes. We may remain somewhat backward compatible by having SetLayoutID(int) method that maps known layout integer IDs to the new string IDs.